### PR TITLE
[18994] Include XTypes v1.3 Annex B, TypeObject IDL and use Fast DDS-Gen to autogenerate code

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
## Description

TL;DR The important files are the IDLs cause the code is mostly auto-generated using Fast DDS-Gen. Once Fast DDS-Gen has been updated the code should be updated. Only modifications related to `@external` annotation should be kept.

This PR includes the TypeObject IDL defined in XTypes v1.3 specification in Annex B. The IDL has been split into several IDLs file to keep the current header structure. The corresponding header and source files have been substituted by the autogenerated files using Fast DDS-Gen. However, some considerations have to be taken into account:

1. Fast DDS-Gen does not allow for the source files to be in a different folder. The source files should be modified to include the proper path to the header for the compiler to found it.
2. Current Fast DDS-Gen has some issues:
    - Aliased bitmasks members in a structures are not generated correctly. This issue is being fixed in Fast DDS-Gen.
    - Empty structures issue unused parameter warnings. This is also going to be tackled in Fast DDS-Gen.
    - Fast DDS-Gen does not support `@annotation`. Thus, even though the Builtin Annotations IDL has been included too, its corresponding TypeObject has not been generated using Fast DDS-Gen. This TypeObject will be updated in a follow-up PR.
    - Fast DDS-Gen does not support `@external`. Therefore, the TypeIdentifierTypes.idl cannot be correctly generated. It has been modified manually.
    - Forward declarations are not correctly supported with `@external` annotations. This causes some warnings issued by Fast DDS-Gen when generating the code. Nevertheless, as the affected code corresponds to the `getMaxSerializedSize` methods which are removed, it is not worrisome at the moment. This issue is propagated to every source file depending on TypeIdentifierTypes.idl.
    - Fast DDS-Gen seems to not be aware when the type is fully defined and it adds TypeIdentifier pointers when the type is already defined and known. This should be fixed in Fast DDS-Gen.
3. MemberFlag and TypeFlag were defined as classes, when the specification states that Bitmasks are to be represented as enumerations. Consequently, the code setting these flags has been commented out until it is refactored.
4. The previous code included some auxiliary methods which are not autogenerated by Fast DDS-Gen. These methods have been removed to improve the maintainability of the code. It should be decided where to include them. They are related to:
    - Type Consistency QoS and checking the compatibility of two different types.
    - `wstring_to_bytes`, `from_string`, and `to_string` helper methods.
6. The previous code had some differences (probably typos) with respect to the IDL defined in the specification. Consequently, some lines have been commented out until those types are refactored.

Every commented out line has been marked with `TODO(jlbueno): XTypes` so it can be fixed in subsequent PRs.

Some of these changes can be considered API breaks, but it can be considered that it was not according to the specification. It can be discussed.

This PR only ensures that Fast DDS library is built. Tests are not going to be built for the moment (test types should be regenerated once Fast DDS-Gen is refactored to support the proper TypeObjects according to the specification).

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [N/A] The code follows the style guidelines of this project. 
- [N/A] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [N/A] Any new/modified methods have been properly documented using Doxygen. 
- [NO] Changes are ABI compatible. 
- [NO] Changes are API compatible. 
- [PENDING] New feature has been added to the `versions.md` file (if applicable).
- [N/A] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [N/A] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
Closing as the previous API is going to be kept and the new is going to be hosted within `fastdds` folder
